include/cadmium/modeling/dynamic_message_bag.hpp

```diff
@@ -0,0 +1,125 @@
+            virtual std::type_index to_type_index() const = 0;
+            virtual std::type_index to_port_type_index() const = 0;
+            virtual void route_messages(const cadmium::dynamic::message_bags& bags_from, cadmium::dynamic::message_bags& bags_to) const = 0;
+        };
+
+        //TODO(Lao): add logger, check if the logger can be made transparently for the user,
+        //f not, the message should be made in the simulator instead.
+        template<typename PORT_FROM, typename PORT_TO>
+        class link : public link_abstract {
+        public:
+            using from_message_type = typename PORT_FROM::message_type;
+            using from_message_bag_type = typename cadmium::message_bag<PORT_FROM>;
+            using to_message_type = typename PORT_TO::message_type;
+            using to_message_bag_type = typename cadmium::message_bag<PORT_TO>;
+
+            link() {
+                static_assert(
+                        std::is_same<from_message_type, to_message_type>::value,
+                        "FROM_PORT message type and TO_PORT message types must be the same type");
+            }
+
+            std::type_index from_type_index() const override {
+                return typeid(from_message_bag_type);
+            }
+
+            std::type_index from_port_type_index() const override {
+                return typeid(PORT_FROM);
+            }
+
+            std::type_index to_type_index() const override {
+                return typeid(to_message_bag_type);
+            }
+
+            std::type_index to_port_type_index() const override {
+                return typeid(PORT_TO);
+            }
+
+            void pass_messages(const boost::any& bag_from, boost::any& bag_to) const {
+                from_message_bag_type b_from = boost::any_cast<from_message_bag_type>(bag_from);
+                to_message_bag_type* b_to = boost::any_cast<to_message_bag_type>(&bag_to);
+                b_to->messages.insert(b_to->messages.end(), b_from.messages.begin(), b_from.messages.end());
+            }
+
+            boost::any pass_messages_to_new_bag(const boost::any& bag_from) const {
+                from_message_bag_type b_from = boost::any_cast<from_message_bag_type>(bag_from);
+                to_message_bag_type b_to;
+                b_to.messages.insert(b_to.messages.end(), b_from.messages.begin(), b_from.messages.end());
+                return b_to;
+            }
+
+            /**
+             * @note This methods assumes the port is defined in the message_bagas parameter bag,
+             * if is not the case, the function throws a std::map out of range exception.
+             *
+             * @param bags - The cadmium::dynamic::message_bags to check if there is messages in the from port
+             * @return true if there is messages, otherwise false
+             */
+            bool there_is_messages_to_route(const cadmium::dynamic::message_bags& bags) const {
```

Comment:
I think it sound better "is_there_messages_to_route". usually convention is to start "boolean gets" with "is"

Reply:
Done, moved from affirmation to question name convention.

Reply:
I usually prefer the affirmation convention because it let the code more like in the natural languaje;

If there is messages to send, then ... 
instead of
If is there messages to send, then ... (which is weird to say)

but is just a convention, I don't really care if there is consistency. I care if is to awful.